below/procfs/src/lib.rs

```diff
@@ -1305,21 +1309,35 @@ impl NetReader {
     }
 
     pub fn read_netstat(&self) -> Result<NetStat> {
-        let netstat_map = self.read_kv_diff_line("netstat")?;
-        let snmp_map = self.read_kv_diff_line("snmp")?;
-        let snmp6_map = self.read_kv_same_line("snmp6")?;
+        let netstat_map = handle_io_error(&self.logger, self.read_kv_diff_line("netstat"))?;
+        let snmp_map = handle_io_error(&self.logger, self.read_kv_diff_line("snmp"))?;
+        let snmp6_map = handle_io_error(&self.logger, self.read_kv_same_line("snmp6"))?;
+        let iface_map = handle_io_error(&self.logger, self.read_net_map())?;
 
         Ok(NetStat {
-            interfaces: Some(self.read_net_map()?),
-            tcp: Some(Self::read_tcp_stat(&snmp_map)),
-            tcp_ext: Some(Self::read_tcp_ext_stat(&netstat_map)),
-            ip: Some(Self::read_ip_stat(&snmp_map)),
-            ip_ext: Some(Self::read_ip_ext_stat(&netstat_map)),
-            ip6: Some(Self::read_ip6_stat(&snmp6_map)),
-            icmp: Some(Self::read_icmp_stat(&snmp_map)),
-            icmp6: Some(Self::read_icmp6_stat(&snmp6_map)),
-            udp: Some(Self::read_udp_stat(&snmp_map)),
-            udp6: Some(Self::read_udp6_stat(&snmp6_map)),
+            interfaces: iface_map,
+            tcp: snmp_map.as_ref().map(Self::read_tcp_stat),
+            tcp_ext: netstat_map.as_ref().map(Self::read_tcp_ext_stat),
+            ip: snmp_map.as_ref().map(Self::read_ip_stat),
+            ip_ext: netstat_map.as_ref().map(Self::read_ip_ext_stat),
+            ip6: snmp6_map.as_ref().map(Self::read_ip6_stat),
+            icmp: snmp_map.as_ref().map(Self::read_icmp_stat),
+            icmp6: snmp6_map.as_ref().map(Self::read_icmp6_stat),
+            udp: snmp_map.as_ref().map(Self::read_udp_stat),
+            udp6: snmp6_map.as_ref().map(Self::read_udp6_stat),
         })
     }
 }
+
+fn handle_io_error<K, V>(
+    logger: &slog::Logger,
+    result: Result<BTreeMap<K, V>>,
+) -> Result<Option<BTreeMap<K, V>>> {
+    let netstat_map = if let Err(Error::IoError(_, err)) = result {
+        warn!(logger, "{:?}", err);
```

Comment:
On a system that's permanently missing the file (like with our ipv6), this will spam the logs, right? If there's a `warn_once!` macro that would probably be better. If not, probably best to drop the logs altogether.

Reply:
couldn't find `warn_once!` in the `slog` crate, however we could replace it with `debug!` instead?

Reply:
Yeah, that sounds reasonable